src/ring_theory/graded_algebra/radical.lean

```diff
@@ -0,0 +1,278 @@
+      specialize not_mem_j H₂,
+      apply ideal.mul_mem_left,
+      convert not_mem_j, },
+    { cases h,
+      { -- in this case `i = max₁`, so `max₂ = j`, contradictory
+        have : j = max₂,
+        { rw h at H₃,
+          exact linear_ordered_cancel_add_comm_monoid.add_left_cancel _ _ _ H₃, },
+        exfalso,
+        exact H₄ h this, },
+      { -- in this case `i > max₁`, so `i < max₁`, so `of A i (x i) ∈ I`
+        have ineq : max₁ < i,
+        { by_contra rid₂, rw not_lt at rid₂,
+          have rid₃ := add_lt_add_of_le_of_lt rid₂ h,
+          conv_lhs at rid₃ { rw linear_ordered_cancel_add_comm_monoid.add_comm },
+          exact lt_irrefl _ rid₃, },
+        have not_mem_i : i ∉ set₁,
+        { intro rid₂,
+          rw max₁_eq at ineq,
+          have rid₃ := (finset.max'_lt_iff set₁ set₁_nonempty).mp ineq i rid₂,
+          exact lt_irrefl _ rid₃,},
+        rw set₁_eq at not_mem_i,
+        simp only [not_and, not_not, ne.def, dfinsupp.mem_support_to_fun,
+          finset.mem_filter] at not_mem_i,
+        specialize not_mem_i H₁,
+        apply ideal.mul_mem_right _ I,
+        convert not_mem_i, }, } },
+  have mem_I₃ :
+    (graded_algebra.proj 𝒜 (max₁, max₂).fst) x * (graded_algebra.proj 𝒜 (max₁, max₂).snd) y ∈ I,
+  { rw eq₂, apply ideal.sub_mem,
+    specialize hI (max₁ + max₂) hxy, exact hxy, exact mem_I₂, },
+  specialize homogeneous_mem_or_mem ⟨max₁, _⟩ ⟨max₂, _⟩ mem_I₃,
+  rw [graded_algebra.proj_apply], exact submodule.coe_mem _,
+  rw [graded_algebra.proj_apply], exact submodule.coe_mem _,
+  cases homogeneous_mem_or_mem,
+  simp only [ne.def, dfinsupp.mem_support_to_fun, finset.mem_filter] at mem_max₁,
+  refine mem_max₁.2 homogeneous_mem_or_mem,
+  simp only [ne.def, dfinsupp.mem_support_to_fun, finset.mem_filter] at mem_max₂,
+  refine mem_max₂.2 homogeneous_mem_or_mem,
+
+  -- (0)
+  { rw [graded_algebra.proj_apply, alg_equiv.map_mul, graded_algebra.support,
+      graded_algebra.support, direct_sum.coe_mul_apply_submodule], refl },
+
+  -- (1)
+  { congr, ext, split; intros H,
+    { simp only [finset.mem_filter, ne.def, dfinsupp.mem_support_to_fun, finset.mem_product] at H,
+      rw finset.mem_union,
+      by_cases a = (max₁, max₂),
+      right, rw h, exact finset.mem_singleton_self (max₁, max₂),
+      left, rw finset.mem_sdiff, split,
+      simp only [finset.mem_filter, ne.def, dfinsupp.mem_support_to_fun, finset.mem_product],
+      exact H, intro rid, simp only [finset.mem_singleton] at rid, exact h rid, },
+    { rw finset.mem_union at H, cases H,
+      rw finset.mem_sdiff at H, exact H.1,
+      simp only [finset.mem_filter, ne.def, dfinsupp.mem_support_to_fun, finset.mem_product],
+      simp only [finset.mem_singleton] at H, rw H,
+      refine ⟨⟨_, _⟩, rfl⟩,
+      simp only [ne.def, dfinsupp.mem_support_to_fun, finset.mem_filter] at mem_max₁,
+      exact mem_max₁.1,
+      simp only [ne.def, dfinsupp.mem_support_to_fun, finset.mem_filter] at mem_max₂,
+      exact mem_max₂.1, }, },
```

Comment:
`finset.sdiff_union_of_subset` ought to be able to mop a lot of this up.

Reply:
indeed